### PR TITLE
Firebase

### PR DESCRIPTION
**Pull request by lyw0149**
intergrated firebase.

### CODE REVIEW

File: src/server/graphQLschema/mutation/user.mutation.js [hunk 1]
```diff
@@ -60,24 +94,28 @@ const UserMutation = {
...
-            if (user) {
-              user.comparePassword(password, (err, isMatch) => {
-                if (isMatch) {
-                  return resolve({ accessToken: jwtUtil.createAccessToken(user) });
-                }
-
-                return reject('Wrong password.');
-              });
+        admin.auth().getUserByEmail(email)
+          .then(function (userRecord) {
```
Comment (ochanje210) [inline src/server/graphQLschema/mutation/user.mutation.js:101]: is there a reason for not having arrow function?

Comment (lyw0149) [reply src/server/graphQLschema/mutation/user.mutation.js:101]: I'll change it

Comment (ochanje210) [reply src/server/graphQLschema/mutation/user.mutation.js:101]: thanks. just wondered if there was a difference

File: src/server/util/ref.util.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+const utils = {
+  usersRef: admin.database().ref('/users'),
+  booksRef: admin.database().ref('/books'),
+  snippetsRef: admin.database().ref('/snippets'),
+  myLibraryRef: admin.database().ref('/userReadable/myLibrary'),
+  savedRef: admin.database().ref('/userReadable/saved'),
+  trashRef: admin.database().ref('/userReadable/snippetTrash'),
+};
+
+export default utils;
```
Comment (ochanje210) [inline src/server/util/ref.util.js]: new line needed

File: package.json [hunk 3]
```diff
@@ -19,14 +19,11 @@
     "body-parser": "^1.15.2",
     "express": "^4.14.0",
     "express-graphql": "^0.6.1",
-    "firebase": "^3.6.4",
```
Comment (ochanje210) [inline package.json]: what was this for?

**Comment by ochanje210** [comment]
overall looks good

**Comment by ochanje210** [comment]
leaving those few minor things to be fixed, I am giving you LGTM

[Approved with PullApprove](https://pullapprove.com/recbook/recbook-server/pull-request/11/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=recbook/recbook-server)

**Pull request merged**